src/services/utilities.ts

```diff
@@ -1661,6 +1661,13 @@ namespace ts {
         return position;
     }
 
+    export function getPrecedingNonSpaceCharacterPosition(text: string, position: number) {

```

Comment:
I find it confusing that this function might return `-1` and you unconditionally use `getPrecedingNonSpaceCharacterPosition(...) + 1` whenever you call it -- maybe this should just be returning one more than it currently does.